drivers/bus/acpi/compbatt/compbatt.h

```diff
@@ -1,156 +1,327 @@
 /*
- * PROJECT:         ReactOS Composite Battery Driver
- * LICENSE:         BSD - See COPYING.ARM in the top level directory
- * FILE:            boot/drivers/bus/acpi/compbatt/compbatt.h
- * PURPOSE:         Main Header File
- * PROGRAMMERS:     ReactOS Portable Systems Group
+ * PROJECT:     ReactOS Composite Battery Driver
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Composite battery main header file
+ * COPYRIGHT:   Copyright 2010 ReactOS Portable Systems Group <ros.arm@reactos.org>
+ *              Copyright 2024 George Bișoc <george.bisoc@reactos.org>
  */
 
+/* INCLUDES *******************************************************************/
+
 #ifndef _COMPBATT_PCH_
 #define _COMPBATT_PCH_
 
 #include <wdm.h>
 #include <batclass.h>
 
+/* DEFINES ********************************************************************/
+
+//
+// I/O remove lock allocate tag
+//
+#define COMPBATT_TAG                            'aBoC'
+
+//
+// Composite battery flags
+//
 #define COMPBATT_BATTERY_INFORMATION_PRESENT    0x04
+#define COMPBATT_STATUS_NOTIFY_SET              0x10
 #define COMPBATT_TAG_ASSIGNED                   0x80
 
+//
+// IRP complete worker mode states
+//
+#define COMPBATT_QUERY_TAG                      1
+#define COMPBATT_READ_STATUS                    2
+
+//
+// Low/High capacity wait constants
+//
+#define COMPBATT_WAIT_MIN_LOW_CAPACITY          0
+#define COMPBATT_WAIT_MAX_HIGH_CAPACITY         0x7FFFFFFF
+
+//
+// One hour in seconds, used to calculate the total rate of each battery for time estimation
+//
+#define COMPBATT_ATRATE_HOUR_IN_SECS            3600
+
+//
+// Time constant of which the battery status data is considered fresh (50000000 * 100ns == 5s)
+//
+#define COMPBATT_FRESH_STATUS_TIME              50000000
+
+//
+// Macro that calculates the delta of a battery's capacity
+//
+#define COMPUTE_BATT_CAP_DELTA(LowDiff, Batt, TotalRate)            \
+    ((ULONG)(((LONGLONG)LowDiff * (Batt)->BatteryStatus.Rate) / TotalRate))
+
+//
+// Macro that calculates the "At Rate" time drain of the battery
+//
+#define COMPUTE_ATRATE_DRAIN(Batt, Time)                            \
+    ((LONG)((Batt)->BatteryStatus.Capacity) * COMPBATT_ATRATE_HOUR_IN_SECS / Time)
+
+//
+// Composite battery debug switches
```

Comment:
```suggestion
// Composite battery debug levels
```
(see "debug levels", "log levels"...)